### DISCUSSION TITLE
How to calculate percentage?

### DISCUSSION THREAD

**Discussion by MarcIrving** [atoti/atoti#582] [Q&A] [answered]
Hi guys please i need a little help , i want to calculate the percentage  of the sum for a specific row in all other a column  ,
let me expain more , for exemple  have 
```
h ["social media"]=
Facebook
instagram
Facebook
Instagram
Facebook
Youtube
```
i want to have the measure as this way 
```
m["Percent %"]=
Facebook 40%
instagram 10%
Youtube 50%
```
**i tried this to calculate the number of occurences of Facebook but it doesnt Work**
`m["Percent %"] = tt.agg.sum( tt.where((h["social"] == "Facebook") `

**Comment by bloussou** [comment, marked as answer]
Hello,
1. What is your percentage? (It must be the "quantity" of something divide by a total). What is the measure representing the quantity in your case?

2. Please read the [`atoti.where()` documentation](https://docs.atoti.io/latest/lib/atoti/atoti.function.where.html#atoti.where). It has an example to show you how to use it.

Some help from what I understand:
```python
m["Total(contributors.COUNT)"] = tt.total(m["contributors.COUNT"], h["social media"]) # the total
m["Percent"] = m["contributors.COUNT"] / m["Total(contributors.COUNT)"]
```
Then display it on a table with social medias on rows.
[`atoti.total()` documentation](https://docs.atoti.io/latest/lib/atoti/atoti.function.total.html#atoti.total)

**Reply by MarcIrving** [reply]
`TypeError: type of argument "hierarchies"[0] must be atoti.hierarchy.Hierarchy; got atoti.column.Column instead`
I got this as an error

**Reply by bloussou** [reply]
The error is showing that you are not respecting the signature of the function. you have probably used `table[...]` (of type `Column`) instead of `h[...]` of type `Hierarchy` where a `Hierarchy` is expected. I cannot give more precision without the full stacktrace

**Reply by MarcIrving** [reply]
its solved i gave the wrong data , @bloussou Let's suppose i want how many time Facebook Appeared , how can i do it ?

**Reply by bloussou** [reply]
something like `m["Facebook.COUNT"] = tt.where(l["social media"] == "Facebook", m["contributors.COUNT"])` should work

**Reply by MarcIrving** [reply]
i mean for all the rest of remainning Media like a measure to calculate them all

**Reply by bloussou** [reply]
then it is just `m["contributors.COUNT"]`

**Reply by MarcIrving** [reply]
Yes i see , but the results will not be like this
```
Facebook 3
Instagram 2
Youtube 1
``` 
its about the number of occurences

**Reply by bloussou** [reply]
why not ? If you do `cube.query(m["contributors.COUNT"], levels=[l["social media"]])` it will exactly show this:
```
                       contributors.COUNT
social media
Facebook                               3
Instagram                              2
Youtube                                1
```

**Reply by MarcIrving** [reply]
YES i know but i'm using the web application so i can't use it directly

**Reply by bloussou** [reply]
can you share a screenshot ? I don't get why you cannot access this measure and level in the web application

**Reply by MarcIrving** [reply]
what i'm trying to say , i'm doing the measures and then i visualize them in the web application , when i do cube query its always in the notebook , not in the web interface , and forgive my bad english

**Reply by bloussou** [reply]
If you are creating a pivot table and choose `social media` on rows and `contributors.COUNT` on column then you have exactly the output of `cube.query()`.
If you want me to correctly understand the issue please provide a screenshot of the web application.

**Reply by MarcIrving** [reply]
@bloussou can you please tell me how can i do it with two inputs like` tt.where(l["social media"] == "Facebook" and "Instagram", m["contributors.COUNT"])`

**Reply by bloussou** [reply]
what you want to do is probably a `or` : `tt.where((l["social media"] == "Facebook") | (l["social media"] == "Instagram"), m["contributors.COUNT"])`

**Reply by MarcIrving** [reply]
i can't combine them ? a `plus` with & maybe ?

**Reply by MarcIrving** [reply]
@bloussou i mean why i can't have the count at the same measure like from : 
```
                   contributors.COUNT
social media
Facebook                               3
Instagram                              2
Youtube                                1 
```
To 
` FacebookAndInstagram.COUNT = 5`
